<a href="https://colab.research.google.com/github/cassianovello/mpv_01/blob/main/DataScience_MVP_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


Data Science - MVP 01



In [131]:
# Importação de pacotes
import pandas as pd
import numpy as np



Leitura dos dados

In [132]:
# url a importar
url_dados = 'https://raw.githubusercontent.com/cassianovello/mpv_01/main/Feline_dataset_compact.csv'

# carga do dataset através da url
df = pd.read_csv(url_dados,sep=';')

df.head()

CATID                    BREEDGROUP     SEX AGE_BEHAVIOUR OTHER_CATS  \
0      1  Siberian_and_Neva_Masquerade  Female   5,079872204    unknown   
1      2          Sphynx_and_Devon_Rex  Female   4,225506294    unknown   
2      3                      Oriental  Female   7,419575633         NO   
3      4          Sphynx_and_Devon_Rex  Female             2        YES   
4      5                     House_cat  Female   3,535049288        YES   

  problematic_behavior   fearfulness human_aggression activity_playfulness  \
0              unknown  -0,592535855     -0,376838532          0,677554498   
1              unknown   0,961022037      1,890785811         -0,585892552   
2             A_little   -0,48003798     -0,288478537          0,583599898   
3                   No  -0,876062946     -0,696218261          1,255441754   
4             A_little  -0,746316221     -0,619939655         -0,578863964   

  cat_sociability human_sociability litterbox_issues excessive_grooming  
0     0,247060591       1,318159592     -0,370579892       -0,329714164  
1     -1,31907745      -1,185782329     -0,758987462        0,986190805  
2       0,1834663       1,313597761     -0,127346636       -0,009533091  
3     0,899893583       0,475639017     -0,514913155       -0,259346702  
4    -0,337497461      -0,333069701      3,084435142       -0,113738586

In [133]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4316 entries, 0 to 4315
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   CATID                 4316 non-null   int64 
 1   BREEDGROUP            4316 non-null   object
 2   SEX                   4316 non-null   object
 3   AGE_BEHAVIOUR         4316 non-null   object
 4   OTHER_CATS            4316 non-null   object
 5   problematic_behavior  4316 non-null   object
 6   fearfulness           4316 non-null   object
 7   human_aggression      4316 non-null   object
 8   activity_playfulness  4316 non-null   object
 9   cat_sociability       4316 non-null   object
 10  human_sociability     4316 non-null   object
 11  litterbox_issues      4316 non-null   object
 12  excessive_grooming    4316 non-null   object
dtypes: int64(1), object(12)
memory usage: 438.5+ KB


Podemos observar que apenas a coluna "CATID" foi interpretada como número. A coluna "age_behaviour" e todas as colunas a partir de "fearfulness" devem ser interpretadas como números de ponto flutuante. Vamos acertar o separador decimal, e realizar a conversão das colunas.

In [134]:
df = df.replace(',','.',regex=True)
float_cols = df.loc[:,'fearfulness':].columns
df[float_cols]= df[float_cols].astype(float)
print(df.dtypes)

CATID                     int64
BREEDGROUP               object
SEX                      object
AGE_BEHAVIOUR            object
OTHER_CATS               object
problematic_behavior     object
fearfulness             float64
human_aggression        float64
activity_playfulness    float64
cat_sociability         float64
human_sociability       float64
litterbox_issues        float64
excessive_grooming      float64
dtype: object


Ainda precisamos tratar a coluna AGE_BEHAVIOUR. Ela possui alguns preenchimentos de texto, que devemos analisar.

In [136]:
float_age = pd.to_numeric(df['AGE_BEHAVIOUR'],errors="coerce")
text_age = df['AGE_BEHAVIOUR'].loc[float_age.isna()]
print("Preenchimentos em texto",text_age.unique())
print("Quantidade de textos na coluna:",float_age.isna().sum())



Preenchimentos em texto ['unknown']
Quantidade de textos na coluna: 123


Apenas 123 dos 4316 registros foram preenchidos com idade "unknown". Por ser uma quantidade pequena no total, iremos desconsiderar estes registros no estudo.

In [126]:
df['AGE_BEHAVIOUR'] = pd.to_numeric(df['AGE_BEHAVIOUR'],errors="coerce")
df.isna().sum()


CATID                     0
BREEDGROUP                0
SEX                       0
AGE_BEHAVIOUR           123
OTHER_CATS                0
problematic_behavior      0
fearfulness               0
human_aggression          0
activity_playfulness      0
cat_sociability           0
human_sociability         0
litterbox_issues          0
excessive_grooming        0
dtype: int64

In [127]:
df = df.dropna()
df

CATID                    BREEDGROUP     SEX  AGE_BEHAVIOUR OTHER_CATS  \
0         1  Siberian_and_Neva_Masquerade  Female       5.079872    unknown   
1         2          Sphynx_and_Devon_Rex  Female       4.225506    unknown   
2         3                      Oriental  Female       7.419576         NO   
3         4          Sphynx_and_Devon_Rex  Female       2.000000        YES   
4         5                     House_cat  Female       3.535049        YES   
...     ...                           ...     ...            ...        ...   
4311   4401                         Korat  Female       0.333333    unknown   
4312   4402                         Korat  Female       0.333333    unknown   
4313   4403                         Korat  Female       0.333333    unknown   
4314   4404                         Korat    Male       0.333333    unknown   
4315   4405        Landrace_Cat_Shorthair    Male       0.334247        YES   

     problematic_behavior  fearfulness  human_aggression  \
0                 unknown    -0.592536         -0.376839   
1                 unknown     0.961022          1.890786   
2                A_little    -0.480038         -0.288479   
3                      No    -0.876063         -0.696218   
4                A_little    -0.746316         -0.619940   
...                   ...          ...               ...   
4311              unknown    -1.196693         -0.315187   
4312              unknown    -1.563574          0.855251   
4313              unknown    -1.515542          0.745532   
4314              unknown    -1.218929         -0.759734   
4315                   No    -1.098438         -0.467702   

      activity_playfulness  cat_sociability  human_sociability  \
0                 0.677554         0.247061           1.318160   
1                -0.585893        -1.319077          -1.185782   
2                 0.583600         0.183466           1.313598   
3                 1.255442         0.899894           0.475639   
4                -0.578864        -0.337497          -0.333070   
...                    ...              ...                ...   
4311              1.305526         0.822008           0.010866   
4312              1.400834         0.559283          -0.376932   
4313              1.412664         0.521732          -0.075399   
4314              1.034383         0.856490           0.106618   
4315              0.431268         0.316437          -1.258882   

      litterbox_issues  excessive_grooming  
0            -0.370580           -0.329714  
1            -0.758987            0.986191  
2            -0.127347           -0.009533  
3            -0.514913           -0.259347  
4             3.084435           -0.113739  
...                ...                 ...  
4311         -0.650696           -0.371469  
4312         -0.630812           -0.374776  
4313         -0.593704           -0.453170  
4314         -0.746913           -0.483224  
4315         -0.987425           -0.459023  

[4193 rows x 13 columns]

In [128]:
df.describe(include='all')

CATID              BREEDGROUP   SEX  AGE_BEHAVIOUR OTHER_CATS  \
count   4193.000000                    4193  4193    4193.000000       4193   
unique          NaN                      26     2            NaN          3   
top             NaN  Landrace_Cat_Shorthair  Male            NaN        YES   
freq            NaN                     968  2163            NaN       2592   
mean    2209.826854                     NaN   NaN       5.826055        NaN   
std     1270.600928                     NaN   NaN       4.133711        NaN   
min        1.000000                     NaN   NaN       0.273973        NaN   
25%     1104.000000                     NaN   NaN       2.269161        NaN   
50%     2220.000000                     NaN   NaN       4.937471        NaN   
75%     3308.000000                     NaN   NaN       8.748631        NaN   
max     4405.000000                     NaN   NaN      22.693333        NaN   

       problematic_behavior  fearfulness  human_aggression  \
count                  4193  4193.000000       4193.000000   
unique                    5          NaN               NaN   
top                      No          NaN               NaN   
freq                   1818          NaN               NaN   
mean                    NaN    -0.004795         -0.002860   
std                     NaN     0.982613          0.976135   
min                     NaN    -1.684218         -1.191298   
25%                     NaN    -0.819918         -0.642711   
50%                     NaN    -0.199770         -0.346189   
75%                     NaN     0.662508          0.297392   
max                     NaN     3.801571          7.018927   

        activity_playfulness  cat_sociability  human_sociability  \
count            4193.000000      4193.000000        4193.000000   
unique                   NaN              NaN                NaN   
top                      NaN              NaN                NaN   
freq                     NaN              NaN                NaN   
mean                0.014571         0.006193          -0.000514   
std                 0.988884         0.981418           0.992559   
min                -4.326500        -4.939943          -5.065313   
25%                -0.557622        -0.402539          -0.531702   
50%                 0.147105         0.168171           0.111480   
75%                 0.738601         0.697493           0.661964   
max                 2.512806         2.098537           2.835573   

        litterbox_issues  excessive_grooming  
count        4193.000000         4193.000000  
unique               NaN                 NaN  
top                  NaN                 NaN  
freq                 NaN                 NaN  
mean           -0.006301           -0.005408  
std             0.987939            0.986189  
min            -1.530227           -1.248026  
25%            -0.631005           -0.523191  
50%            -0.295858           -0.285649  
75%             0.260741            0.083759  
max             6.674081            7.806817